load-test-framework/src/main/java/com/google/pubsub/flic/output/SheetsService.java

```diff
@@ -0,0 +1,140 @@
+import com.google.api.client.http.HttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.client.util.store.FileDataStoreFactory;
+import com.google.api.services.sheets.v4.Sheets;
+import com.google.api.services.sheets.v4.SheetsScopes;
+import com.google.api.services.sheets.v4.model.ValueRange;
+import com.google.pubsub.flic.common.LatencyDistribution;
+import com.google.pubsub.flic.controllers.Controller;
+import com.google.pubsub.flic.controllers.GCEController;
+import com.google.pubsub.flic.controllers.Client;
+import com.google.pubsub.flic.controllers.Client.ClientType;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.LongStream;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class SheetsService {
+  private final static Logger log = LoggerFactory.getLogger(SheetsService.class);
+  private final Sheets service;
+  private int cpsPublisherCount = 0;
+  private int cpsSubscriberCount = 0;
+  private int kafkaPublisherCount = 0;
+  private int kafkaSubscriberCount = 0;
+  
+  private final String APPLICATION_NAME = "loadtest-framework";
+  
+  public SheetsService(String dataStoreDirectory) throws Exception {
+    InputStream in = new FileInputStream(new File(System.getenv("GOOGLE_OATH2_CREDENTIALS")));
+    JsonFactory factory = new JacksonFactory();
+    GoogleClientSecrets clientSecrets = GoogleClientSecrets.load(factory, new InputStreamReader(in));
+    HttpTransport transport = GoogleNetHttpTransport.newTrustedTransport(); 
+    FileDataStoreFactory dataStoreFactory = new FileDataStoreFactory(new File(dataStoreDirectory));
+    List<String> scopes = Arrays.asList(SheetsScopes.SPREADSHEETS);
+    GoogleAuthorizationCodeFlow flow =
+        new GoogleAuthorizationCodeFlow.Builder(
+                transport, factory, clientSecrets, scopes)
+        .setAccessType("offline")
+        .setDataStoreFactory(dataStoreFactory)
+        .build();
+    Credential credential = new AuthorizationCodeInstalledApp(
+        flow, new LocalServerReceiver()).authorize("user");
+    service = new Sheets.Builder(transport, factory, credential)
+        .setApplicationName(APPLICATION_NAME).build();
+  }
+  
+  /* Publishes stats information to Google Sheets document. Format for sheet assumes the following
+   * column order: Publisher #; Subscriber #; Message size (B); Test length (s); # messages;
+   * Publish batch size; Subscribe pull size; Request rate; Max outstanding requests; 
+   * Throughput (MB/s); 50% (ms); 90% (ms); 99% (ms)
+   */
+  public void sendToSheets(String sheetId, Map<ClientType, Controller.LoadtestStats> results,
```

Comment:
Separate into a function that outputs the values, and a function that writes to Sheets. Then please write a unit test for the former. This code is going to get changed with every new type, and it would be good to catch errors in our tests.